ui/src/core/xibo-calendar.js

```diff
@@ -40,6 +40,133 @@ $(document).ready(function() {
         });
     });
 
+    // Set a listener for type change event
+    $('body').on('change', 'select[name="criteria_type[]"]', function() {
```

Comment:
Should we think about scoping this to the add/edit form? As it stands any select field with this name is going to be bound.